policy/lib/rule_data_test.rego

```diff
@@ -26,6 +26,41 @@ test_rule_data if {
 		with lib.rule_data_defaults as {"key3": 10}
 }
 
+test_rule_data_effective_on if {
+	lib.assert_equal(
+		[
+			40, # key0 value comes from data.rule_data__configuration__
+			30, # key1 value comes from data.rule_data_custom
+			20, # key2 value comes from data.rule_data and is effective
+			[], # key3 value comes from lib.rule_data_defaults, but is not yet effective
+			[], # key4 value is not defined
+		],
+		[
+			lib.rule_data("key0"),
+			lib.rule_data("key1"),
+			lib.rule_data("key2"),
+			lib.rule_data("key3"),
+			lib.rule_data("key4"),
+		],
+	) with data.rule_data__configuration__ as {"key0": 40}
+		with data.rule_data_custom as {"key0": 30, "key1": 30}
+		with data.rule_data as {
+			"key0": {
+				"value": 20,
+				"effective_on": "2024-01-01T00:00:00Z",
+			},
+			"key1": {
+				"value": 20,
+				"effective_on": "2024-01-01T00:00:00Z",
+			},
+			"key2": {
+				"value": 20,
+				"effective_on": "2024-01-01T00:00:00Z",
+			},
+		}
+		with lib.rule_data_defaults as {"key3": {"value": 10, "effective_on": "9999-01-01T00:00:00Z"}}
+}
+
```

Comment:
This test seems odd since it appears to want to retest the rule data precedence stuff already covered in the previous test.

Maybe redesign it so we're just testing the effective_on behavior, e.g.:

```
   # Effective now
   assert_equal("bar", data.rule_data("foo")) with data.rule_data as { "foo": { "value": "bar", "effective_on": "..." }

   # Not effective yet
   assert_equal([], data.rule_data("foo")) with data.rule_data as { "foo": { "value": "bar", "effective_on": "..." }

```